### ISSUE TITLE
Make error

### ISSUE THREAD

**Issue by MartyME91** [Drumfix/motu-avb-usb#7]
Hi Drumfix,

Thanks for providing this code! I have a MOTU 24Ao in the lab and am trying to control it from a PC running Ubuntu 20.04. I am also experiencing some issues with it suddenly stopping in the middle of playback and channels switching (especially when using an USB extension cable between sound card and PC) and I was hoping this driver would fix that. However, I have some problems compiling and installing the driver and was hoping you could help. I have only a little programming experience, so sorry if this is a trivial question. 

So, I installed the kernel source code by downloading it from https://github.com/zen-kernel/zen-kernel/tags and extracting them into/lib/modules/`uname -r`/source as @ozel suggested. But when I try to run `make` I get the following:
_Copying required header files from kernel sources...
'/lib/modules/5.11.0-41-generic/source/sound/usb/usbaudio.h' -> '/home/audiologie/Documents/motu-avb-usb-main/usbaudio.h'
'/lib/modules/5.11.0-41-generic/source/sound/usb/midi.h' -> '/home/audiologie/Documents/motu-avb-usb-main/midi.h'
make -C ""/lib/modules/5.11.0-41-generic/build"" M="/home/audiologie/Documents/motu-avb-usb-main" modules
make[1]: Entering directory '/usr/src/linux-headers-5.11.0-41-generic'
  CC [M]  /home/audiologie/Documents/motu-avb-usb-main/motu.o
  MODPOST /home/audiologie/Documents/motu-avb-usb-main/Module.symvers
  LD [M]  /home/audiologie/Documents/motu-avb-usb-main/motu.ko
  BTF [M] /home/audiologie/Documents/motu-avb-usb-main/motu.ko
Skipping BTF generation for /home/audiologie/Documents/motu-avb-usb-main/motu.ko due to unavailability of vmlinux
make[1]: Leaving directory '/usr/src/linux-headers-5.11.0-41-generic'
_

I don't know if this is an error or just a warning that can be ignored, but if I then try to do `sudo make install` I get the following errors:
_make -C ""/lib/modules/5.11.0-41-generic/build"" M="/home/audiologie/Documents/motu-avb-usb-main" modules_install
make[1]: Entering directory '/usr/src/linux-headers-5.11.0-41-generic'
  INSTALL /home/audiologie/Documents/motu-avb-usb-main/motu.ko
At main.c:160:
- SSL error:02001002:system library:fopen:No such file or directory: ../crypto/bio/bss_file.c:69
- SSL error:2006D080:BIO routines:BIO_new_file:no such file: ../crypto/bio/bss_file.c:76
sign-file: certs/signing_key.pem: No such file or directory
  DEPMOD  5.11.0-41-generic
Warning: modules_install: missing 'System.map' file. Skipping depmod.
make[1]: Leaving directory '/usr/src/linux-headers-5.11.0-41-generic'
depmod -A
_

I really have no clue how to fix this, could you please help?

Best, Maartje

**Comment by Drumfix** [comment]
Apparently your system allows only signed modules to be added.


 

Can you load it by:

 

sudo modprobe snd-usbmidi-lib

sudo insmod ./motu.ko

 

from the directory the motu.ko was built?

 

Guess i have to make a dkms module.

 

Gesendet: Dienstag, 30. November 2021 um 12:40 Uhr
Von: "MartyME91" ***@***.***>
An: "Drumfix/motu-avb-usb" ***@***.***>
Cc: "Subscribed" ***@***.***>
Betreff: [Drumfix/motu-avb-usb] Make error (Issue #7)


 

Hi Drumfix,

Thanks for providing this code! I have a MOTU 24Ao in the lab and am trying to control it from a PC running Ubuntu 20.04. I am also experiencing some issues with it suddenly stopping in the middle of playback and channels switching (especially when using an USB extension cable between sound card and PC) and I was hoping this driver would fix that. However, I have some problems compiling and installing the driver and was hoping you could help. I have only a little programming experience, so sorry if this is a trivial question.

So, I installed the kernel source code by downloading it from https://github.com/zen-kernel/zen-kernel/tags and extracting them into/lib/modules/uname -r/source as @ozel suggested. But when I try to run make I get the following:
_Copying required header files from kernel sources...
'/lib/modules/5.11.0-41-generic/source/sound/usb/usbaudio.h' -> '/home/audiologie/Documents/motu-avb-usb-main/usbaudio.h'
'/lib/modules/5.11.0-41-generic/source/sound/usb/midi.h' -> '/home/audiologie/Documents/motu-avb-usb-main/midi.h'
make -C ""/lib/modules/5.11.0-41-generic/build"" M="/home/audiologie/Documents/motu-avb-usb-main" modules
make[1]: Entering directory '/usr/src/linux-headers-5.11.0-41-generic'
CC [M] /home/audiologie/Documents/motu-avb-usb-main/motu.o
MODPOST /home/audiologie/Documents/motu-avb-usb-main/Module.symvers
LD [M] /home/audiologie/Documents/motu-avb-usb-main/motu.ko
BTF [M] /home/audiologie/Documents/motu-avb-usb-main/motu.ko
Skipping BTF generation for /home/audiologie/Documents/motu-avb-usb-main/motu.ko due to unavailability of vmlinux
make[1]: Leaving directory '/usr/src/linux-headers-5.11.0-41-generic'
_

I don't know if this is an error or just a warning that can be ignored, but if I then try to do sudo make install I get the following errors:
_make -C ""/lib/modules/5.11.0-41-generic/build"" M="/home/audiologie/Documents/motu-avb-usb-main" modules_install
make[1]: Entering directory '/usr/src/linux-headers-5.11.0-41-generic'
INSTALL /home/audiologie/Documents/motu-avb-usb-main/motu.ko
At main.c:160:


	SSL error:02001002:system library:fopen:No such file or directory: ../crypto/bio/bss_file.c:69
	SSL error:2006D080:BIO routines:BIO_new_file:no such file: ../crypto/bio/bss_file.c:76
	sign-file: certs/signing_key.pem: No such file or directory
	DEPMOD 5.11.0-41-generic
	Warning: modules_install: missing 'System.map' file. Skipping depmod.
	make[1]: Leaving directory '/usr/src/linux-headers-5.11.0-41-generic'
	depmod -A
	_


I really have no clue how to fix this, could you please help?

Best, Maartje

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by MartyME91** [comment]
> Can you load it by:
> sudo modprobe snd-usbmidi-lib
> sudo insmod  ./motu.ko

No, unfortunately the second command gives the error "ERROR: could not insert module ./motu.ko: Operation not permitted".
It would be great if you could make a dkms module! Is that a lot of work?

**Comment by Drumfix** [comment]
Are you using a dual boot with windows ? In that case make sure to temporarily disable secure boot in the BIOS for the installation to succeed.

**Comment by MartyME91** [comment]
Thank you for your advice. I disabled the secure boot and still can't do the normal `sudo make install`, but doing `sudo modprobe snd-usbmidi-lib` and `sudo insmod ./motu.ko` do not result in errors now. When I do `lsusb -t` it says "Driver=snd-motu-avb", so it seems to work! Unfortunately the issues with playback stopping and wrong channel order are still present though :-( so maybe the problem was the usb extension cable after all.

**Comment by MartyME91** [comment]
Issues also occur without the usb extension cable, only less often. So it's not that, it only makes it worse. I don't know.
